apple silicon support?

Hi, I want to test this library with Mac Mini(m1), but it doesn't work.

``` shell
deno run -Ar --unstable https://deno.land/x/webview/examples/local.ts
...
...
Download https://deno.land/std@0.79.0/hash/_wasm/wasm.js
Download https://deno.land/std@0.79.0/encoding/base64.ts
Download https://deno.land/std@0.79.0/path/_util.ts
Download https://deno.land/std@0.79.0/path/_constants.ts
Download https://deno.land/x/cache@0.2.8/file_fetcher.ts
Check https://deno.land/x/webview/examples/local.ts
error: Uncaught (in promise) Error: Could not open library: dlopen(/Users/paosder-mini/Library/Caches/deno/plug/https/github.com/4b121ee5a08490d714e3f7d99bf1bcce7bafae851df6e895a068bf3e7d1866ae.dylib, 5): no suitable image found.  Did find:
	/Users/paosder-mini/Library/Caches/deno/plug/https/github.com/4b121ee5a08490d714e3f7d99bf1bcce7bafae851df6e895a068bf3e7d1866ae.dylib: mach-o, but wrong architecture
	/Users/paosder-mini/Library/Caches/deno/plug/https/github.com/4b121ee5a08490d714e3f7d99bf1bcce7bafae851df6e895a068bf3e7d1866ae.dylib: stat() failed with errno=35
    at processResponse (deno:core/core.js:223:11)
    at Object.jsonOpSync (deno:core/core.js:246:12)
    at Object.openPlugin (deno:cli/rt/40_plugins.js:7:17)
    at Module.prepare (https://deno.land/x/plug@0.2.6/plug.ts:82:15)
    at async load (https://deno.land/x/webview@0.5.5/plugin.ts:75:9)
    at async https://deno.land/x/webview@0.5.5/mod.ts:5:1
```

It seems doesn't support ARM(m1) yet, however Deno support m1 latest release v1.6 using latest rust to build with.
How do I solve this problem or is there any plan of supporting arm architecture?

Thank you :)

I attempted to build `webview_deno` at version `0.5.6` on my M1 Mac Mini. Build failure report is attached. 

```shell
$ deno run --unstable -A scripts/build.ts
building rust
   Compiling webview-sys v0.6.1 (https://github.com/Boscop/web-view?rev=1c20b92#1c20b925)
   Compiling thiserror-impl v1.0.20
   Compiling futures-macro v0.3.12
   Compiling serde_derive v1.0.123
   Compiling pin-project-internal v1.0.5
   Compiling deno_json_op v0.1.1
The following warnings were emitted during compilation:

warning: webview_cocoa.c:89:23: error: too many arguments to function call, expected 0, have 3
warning:   return objc_msgSend((id)objc_getClass("NSString"),
warning:          ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:95:20: error: too many arguments to function call, expected 0, have 2
warning:       objc_msgSend((id)objc_getClass("NSMenuItem"), sel_registerName("alloc"));
warning:       ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:96:16: error: too many arguments to function call, expected 0, have 5
warning:   objc_msgSend(item, sel_registerName("initWithTitle:action:keyEquivalent:"),
warning:   ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:98:16: error: too many arguments to function call, expected 0, have 2
warning:   objc_msgSend(item, sel_registerName("autorelease"));
warning:   ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:119:27: error: too many arguments to function call, expected 0, have 11
warning:   id event = objc_msgSend((id)objc_getClass("NSEvent"),
warning:              ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:124:25: error: too many arguments to function call, expected 0, have 2
warning:   id app = objc_msgSend((id)objc_getClass("NSApplication"),
warning:            ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:135:34: error: too many arguments to function call, expected 0, have 2
warning:                     objc_msgSend((id)objc_getClass("NSDate"),
warning:                     ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:148:45: error: too many arguments to function call, expected 0, have 2
warning:                                objc_msgSend(message, sel_registerName("body")),
warning:                                ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:155:31: error: too many arguments to function call, expected 0, have 2
warning:   id openPanel = objc_msgSend((id)objc_getClass("NSOpenPanel"),
warning:                  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:160:20: error: too many arguments to function call, expected 0, have 2
warning:       objc_msgSend(parameters, sel_registerName("allowsMultipleSelection")));
warning:       ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:162:16: error: too many arguments to function call, expected 0, have 3
warning:   objc_msgSend(openPanel, sel_registerName("setCanChooseFiles:"), 1);
warning:   ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:166:42: error: too many arguments to function call, expected 0, have 2
warning:           completionHandler(objc_msgSend(openPanel, sel_registerName("URLs")));
warning:                             ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:164:7: error: too many arguments to function call, expected 0, have 3
warning:       openPanel, sel_registerName("beginWithCompletionHandler:"), ^(id result) {
warning:       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:176:31: error: too many arguments to function call, expected 0, have 2
warning:   id savePanel = objc_msgSend((id)objc_getClass("NSSavePanel"),
warning:                  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:178:16: error: too many arguments to function call, expected 0, have 3
warning:   objc_msgSend(savePanel, sel_registerName("setCanCreateDirectories:"), 1);
warning:   ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:179:16: error: too many arguments to function call, expected 0, have 3
warning:   objc_msgSend(savePanel, sel_registerName("setNameFieldStringValue:"),
warning:   ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:184:42: error: too many arguments to function call, expected 0, have 2
warning:                    id url = objc_msgSend(savePanel, sel_registerName("URL"));
warning:                             ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:185:43: error: too many arguments to function call, expected 0, have 2
warning:                    id path = objc_msgSend(url, sel_registerName("path"));
warning:                              ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: webview_cocoa.c:181:16: error: too many arguments to function call, expected 0, have 3
warning:   objc_msgSend(savePanel, sel_registerName("beginWithCompletionHandler:"),
warning:   ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
warning: objc_msgSend(void /* id self, SEL op, ... */ )
warning: ^
warning: fatal error: too many errors emitted, stopping now [-ferror-limit=]
warning: 20 errors generated.

error: failed to run custom build command for `webview-sys v0.6.1 (https://github.com/Boscop/web-view?rev=1c20b92#1c20b925)`

Caused by:
  process didn't exit successfully: `/Users/acs/foss/webview_deno/target/release/build/webview-sys-1807b0e9190e333e/build-script-build` (exit code: 1)
  --- stdout
  cargo:rustc-link-lib=framework=Cocoa
  cargo:rustc-link-lib=framework=WebKit
  TARGET = Some("aarch64-apple-darwin")
  OPT_LEVEL = Some("3")
  HOST = Some("aarch64-apple-darwin")
  CC_aarch64-apple-darwin = None
  CC_aarch64_apple_darwin = None
  HOST_CC = None
  CC = None
  CFLAGS_aarch64-apple-darwin = None
  CFLAGS_aarch64_apple_darwin = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = None
  CC_aarch64-apple-darwin = None
  CC_aarch64_apple_darwin = None
  HOST_CC = None
  CC = None
  CFLAGS_aarch64-apple-darwin = None
  CFLAGS_aarch64_apple_darwin = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = None
  CC_aarch64-apple-darwin = None
  CC_aarch64_apple_darwin = None
  HOST_CC = None
  CC = None
  CFLAGS_aarch64-apple-darwin = None
  CFLAGS_aarch64_apple_darwin = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = None
  running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-arch" "arm64" "-I" "webview.h" "-Wall" "-Wextra" "-x" "objective-c" "-std=c11" "-w" "-DDEBUG" "-DOBJC_OLD_DISPATCH_PROTOTYPES=1" "-o" "/Users/acs/foss/webview_deno/target/release/build/webview-sys-6dd0612f15102572/out/webview_cocoa.o" "-c" "webview_cocoa.c"
  cargo:warning=webview_cocoa.c:89:23: error: too many arguments to function call, expected 0, have 3
  cargo:warning=  return objc_msgSend((id)objc_getClass("NSString"),
  cargo:warning=         ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:95:20: error: too many arguments to function call, expected 0, have 2
  cargo:warning=      objc_msgSend((id)objc_getClass("NSMenuItem"), sel_registerName("alloc"));
  cargo:warning=      ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:96:16: error: too many arguments to function call, expected 0, have 5
  cargo:warning=  objc_msgSend(item, sel_registerName("initWithTitle:action:keyEquivalent:"),
  cargo:warning=  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:98:16: error: too many arguments to function call, expected 0, have 2
  cargo:warning=  objc_msgSend(item, sel_registerName("autorelease"));
  cargo:warning=  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:119:27: error: too many arguments to function call, expected 0, have 11
  cargo:warning=  id event = objc_msgSend((id)objc_getClass("NSEvent"),
  cargo:warning=             ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:124:25: error: too many arguments to function call, expected 0, have 2
  cargo:warning=  id app = objc_msgSend((id)objc_getClass("NSApplication"),
  cargo:warning=           ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:135:34: error: too many arguments to function call, expected 0, have 2
  cargo:warning=                    objc_msgSend((id)objc_getClass("NSDate"),
  cargo:warning=                    ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:148:45: error: too many arguments to function call, expected 0, have 2
  cargo:warning=                               objc_msgSend(message, sel_registerName("body")),
  cargo:warning=                               ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:155:31: error: too many arguments to function call, expected 0, have 2
  cargo:warning=  id openPanel = objc_msgSend((id)objc_getClass("NSOpenPanel"),
  cargo:warning=                 ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:160:20: error: too many arguments to function call, expected 0, have 2
  cargo:warning=      objc_msgSend(parameters, sel_registerName("allowsMultipleSelection")));
  cargo:warning=      ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:162:16: error: too many arguments to function call, expected 0, have 3
  cargo:warning=  objc_msgSend(openPanel, sel_registerName("setCanChooseFiles:"), 1);
  cargo:warning=  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:166:42: error: too many arguments to function call, expected 0, have 2
  cargo:warning=          completionHandler(objc_msgSend(openPanel, sel_registerName("URLs")));
  cargo:warning=                            ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:164:7: error: too many arguments to function call, expected 0, have 3
  cargo:warning=      openPanel, sel_registerName("beginWithCompletionHandler:"), ^(id result) {
  cargo:warning=      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:176:31: error: too many arguments to function call, expected 0, have 2
  cargo:warning=  id savePanel = objc_msgSend((id)objc_getClass("NSSavePanel"),
  cargo:warning=                 ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:178:16: error: too many arguments to function call, expected 0, have 3
  cargo:warning=  objc_msgSend(savePanel, sel_registerName("setCanCreateDirectories:"), 1);
  cargo:warning=  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:179:16: error: too many arguments to function call, expected 0, have 3
  cargo:warning=  objc_msgSend(savePanel, sel_registerName("setNameFieldStringValue:"),
  cargo:warning=  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:184:42: error: too many arguments to function call, expected 0, have 2
  cargo:warning=                   id url = objc_msgSend(savePanel, sel_registerName("URL"));
  cargo:warning=                            ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:185:43: error: too many arguments to function call, expected 0, have 2
  cargo:warning=                   id path = objc_msgSend(url, sel_registerName("path"));
  cargo:warning=                             ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=webview_cocoa.c:181:16: error: too many arguments to function call, expected 0, have 3
  cargo:warning=  objc_msgSend(savePanel, sel_registerName("beginWithCompletionHandler:"),
  cargo:warning=  ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/objc/message.h:63:1: note: 'objc_msgSend' declared here
  cargo:warning=objc_msgSend(void /* id self, SEL op, ... */ )
  cargo:warning=^
  cargo:warning=fatal error: too many errors emitted, stopping now [-ferror-limit=]
  cargo:warning=20 errors generated.
  exit code: 1

  --- stderr


  error occurred: Command "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-arch" "arm64" "-I" "webview.h" "-Wall" "-Wextra" "-x" "objective-c" "-std=c11" "-w" "-DDEBUG" "-DOBJC_OLD_DISPATCH_PROTOTYPES=1" "-o" "/Users/acs/foss/webview_deno/target/release/build/webview-sys-6dd0612f15102572/out/webview_cocoa.o" "-c" "webview_cocoa.c" with args "cc" did not execute successfully (status code exit code: 1).


warning: build failed, waiting for other jobs to finish...
error: build failed
building rust failed
```

--- a/script/build.ts
+++ b/script/build.ts
@@ -1,16 +1,7 @@
 import { ensureDir } from "https://deno.land/std@0.145.0/fs/ensure_dir.ts";
 
 const decoder = new TextDecoder();
-
-const archMap: { [k in typeof Deno.build.arch]: string } = {
-  "x86_64": "x86_64",
-  "aarch64": "arm64",
-} as const;
-
-function indent(source: string, spaces = 2): string {
-  return source.split("\n").map((line) => `${" ".repeat(spaces)}${line}\n`)
-    .join("");
-}
+const architectures = [["x86_64", "x86_64"], ["aarch64", "arm64"]] as const;
 
 const ExitType = {
   Exit: "exit",
@@ -19,9 +10,22 @@ const ExitType = {
 } as const;
 type ExitType = typeof ExitType[keyof typeof ExitType];
 
+const LogType = {
+  Success: "success",
+  Always: "always",
+  Fail: "fail",
+  Never: "never",
+} as const;
+type LogType = typeof LogType[keyof typeof LogType];
+
+function indent(source: string, spaces = 2): string {
+  return source.split("\n").map((line) => `${" ".repeat(spaces)}${line}\n`)
+    .join("");
+}
+
 async function spawn<T extends Deno.SpawnOptions>(
   cmd: string,
-  { opts, exit }: { opts?: T; exit?: ExitType } = { exit: ExitType.Never },
+  { opts, exit, log }: { opts?: T; exit?: ExitType; log?: LogType } = {},
 ): Promise<{
   status: Deno.ChildStatus;
   stdout: string;
@@ -32,31 +36,55 @@ async function spawn<T extends Deno.SpawnOptions>(
     opts.stderr = "piped";
   }
 
-  const { status, stdout, stderr } = await Deno.spawn(cmd, opts);
+  exit ??= ExitType.Never;
+  log ??= LogType.Always;
 
-  if (status.success) {
-    console.log(`Successfully ran "${cmd} ${(opts?.args ?? []).join(" ")}"`);
-    console.log(`stdout:\n${indent(decoder.decode(stdout!))}`);
-    console.log(`stderr:\n${indent(decoder.decode(stderr!))}`);
+  const result = await Deno.spawn(cmd, opts);
+
+  const stdout = decoder.decode(result.stdout!);
+  const stderr = decoder.decode(result.stderr!);
+
+  if (result.status.success) {
+    if (log !== "never") {
+      console.log(`Successfully ran "${cmd} ${(opts?.args ?? []).join(" ")}"`);
+    }
+
+    if (log === "success" || log === "always") {
+      if (stdout.length !== 0) {
+        console.log(`stdout:\n${indent(stdout)}`);
+      }
+      if (stderr.length !== 0) {
+        console.log(`stderr:\n${indent(stderr)}`);
+      }
+    }
   } else {
-    console.log(`Failed run "${cmd}"`);
-    console.log(`stdout:\n${indent(decoder.decode(stdout!))}`);
-    console.log(`stderr:\n${indent(decoder.decode(stderr!))}`);
-    console.log(`status: ${status.code}`);
+    if (log !== "never") {
+      console.log(`Failed run "${cmd}"`);
+    }
+
+    if (log === "fail" || log === "always") {
+      if (stdout.length !== 0) {
+        console.log(`stdout:\n${indent(stdout)}`);
+      }
+      if (stderr.length !== 0) {
+        console.log(`stderr:\n${indent(stderr)}`);
+      }
+      console.log(`status: ${result.status.code}`);
+    }
 
     if (exit === ExitType.Fail) {
-      Deno.exit(status.code);
+      Deno.exit(result.status.code);
     }
   }
 
   if (exit === ExitType.Exit) {
-    Deno.exit(status.code);
+    Deno.exit(result.status.code);
   }
 
   return {
-    status,
-    stdout: decoder.decode(stdout!),
-    stderr: decoder.decode(stderr!),
+    status: result.status,
+    stdout,
+    stderr,
   };
 }
 
@@ -71,30 +99,29 @@ switch (Deno.build.os) {
   }
 
   case "darwin": {
-    await spawn("c++", {
-      opts: {
-        exit: ExitType.Fail,
-        args: [
-          "webview/webview.cc",
-          "-dynamiclib",
-          "-fpic",
-          "-DWEBVIEW_COCOA",
-          "-std=c++11",
-          "-Wall",
-          "-Wextra",
-          "-pedantic",
-          "-framework",
-          "WebKit",
-          "-arch",
-          archMap[Deno.build.arch],
-          "-o",
-          `build/libwebview.${Deno.build.arch}.dylib`,
-        ],
-        env: {
-          "CFLAGS": "",
+    for (const [denoArch, gccArch] of architectures) {
+      await spawn("c++", {
+        opts: {
+          exit: ExitType.Fail,
+          args: [
+            "webview/webview.cc",
+            "-dynamiclib",
+            "-fpic",
+            "-DWEBVIEW_COCOA",
+            "-std=c++11",
+            "-Wall",
+            "-Wextra",
+            "-pedantic",
+            "-framework",
+            "WebKit",
+            "-arch",
+            gccArch,
+            "-o",
+            `build/libwebview.${denoArch}.dylib`,
+          ],
         },
-      },
-    });
+      });
+    }
     Deno.exit(0);
     break;
   }

